tensorflow_lite_support/cc/task/vision/utils/image_tensor_specs.cc

```diff
@@ -39,29 +39,38 @@ using ::tflite::support::StatusOr;
 using ::tflite::support::TfLiteSupportStatus;
 using ::tflite::task::core::TfLiteEngine;
 
-StatusOr<const TensorMetadata*> GetInputTensorMetadataIfAny(
-    const ModelMetadataExtractor& metadata_extractor) {
+StatusOr<const TensorMetadata*> GetTensorMetadataIfAny(
+    const ModelMetadataExtractor& metadata_extractor, const bool is_input) {
   if (metadata_extractor.GetModelMetadata() == nullptr ||
       metadata_extractor.GetModelMetadata()->subgraph_metadata() == nullptr) {
     // Some models have no metadata at all (or very partial), so exit early.
     return nullptr;
-  } else if (metadata_extractor.GetInputTensorCount() != 1) {
+  } else if (is_input && metadata_extractor.GetInputTensorCount() != 1) {
     return CreateStatusWithPayload(
         StatusCode::kInvalidArgument,
         "Models are assumed to have a single input TensorMetadata.",
         TfLiteSupportStatus::kInvalidNumInputTensorsError);
+  } else if (!is_input && metadata_extractor.GetOutputTensorCount() != 1) {
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        "Models are assumed to have a single output TensorMetadata.",
+        TfLiteSupportStatus::kInvalidNumOutputTensorsError);
```

Comment:
`metadata_extractor.GetOutputTensorCount()` returns the output tensor count of the tflite model, which can be greater than one. For example, a model may takes two input image tensors. You can bind an ImagePreprocessor to each of the input image tensor, respectively, but not both input tensors to one ImagePreprocessor (that's what we've agreed in ImagePre(/Post)processor). However, the code will fail here, because `metadata_extractor.GetOutputTensorCount()` returns 2. 

This is a bug in our existing code, it's not discovered before, because all models we supported so far have only one input tensor of image. But we need to correct the logic.

Reply:
And if you pass in `TensorMetadata`, it will be garnered that the number of tensor is 1 for the ImagePostprocessor.